armi/reactor/converters/axialExpansion/assemblyAxialLinkage.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2024 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from armi import runLog
+from armi.reactor.flags import Flags
+from armi.reactor.components import UnshapedComponent
+from armi.reactor.converters.axialExpansion import getSolidComponents
+
+
+class AssemblyAxialLinkage:
+    """Determines and stores the block- and component-wise axial linkage for an assembly.
+
+    Attributes
+    ----------
+    a : :py:class:`Assembly <armi.reactor.assemblies.Assembly>`
+        reference to original assembly; is directly modified/changed during expansion.
+
+    linkedBlocks : dict
+        keys   --> :py:class:`Block <armi.reactor.blocks.Block>`
+
+        values --> list of axially linked blocks; index 0 = lower linked block; index 1: upper linked block.
+
+        see also: self._getLinkedBlocks()
+
+    linkedComponents : dict
+        keys -->   :py:class:`Component <armi.reactor.components.component.Component>`
+
+        values --> list of axially linked components; index 0 = lower linked component; index 1: upper linked component.
+
+        see also: self._getLinkedComponents
```

Comment:
```suggestion
    a : :py:class:`Assembly <armi.reactor.assemblies.Assembly>`
        reference to original assembly; is directly modified/changed during expansion.
    linkedBlocks : dict
        keys   --> :py:class:`Block <armi.reactor.blocks.Block>`
        values --> list of axially linked blocks; index 0 = lower linked block; index 1: upper linked block.
        see also: self._getLinkedBlocks()
    linkedComponents : dict
        keys -->   :py:class:`Component <armi.reactor.components.component.Component>`
        values --> list of axially linked components; index 0 = lower linked component; index 1: upper linked component.
        see also: self._getLinkedComponents
```

Reply:
At your discretion, but for whatever reason, I found this hard to read with all the whitespace.